Fix Translation of error message on cart for deleted bundle option.

swnsma opened the pull request:
### Description
- fix not working translation on Cart Page for error message.

### Manual testing scenarios
#### Pre-conditions
1. Enabled Inline Translates.
2. Bundle product #P1 - visible and saleable.
3. Enabled multishipping.
#### Steps To Reproduce
1. Go to Frontend.
2. Add bundle product to cart.
3. Go to cart page.
4. Go to admin panel.
5. Find bundle product #P1 in Products -> Catalog.
6. Edit options (e.g. change order of items) and save the product.
7. Flush cache (System -> Tools -> Cache Management -> Flush Magento Cache).
8. Go to frontend.
9. Refresh the page.

#### Expected Result:
1. Error messages are shown.
2. All messages are translatable.

#### Actual Result:
1. Error messages are shown.
2. The message 'The required options you selected are not available.' is not inline translatable.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
magento/magento2#?: Fix Translation of error message on cart for deleted bundle option.

magento-engcom-team commented:
Hi @swnsma. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

orlangur commented on app/code/Magento/Multishipping/etc/frontend/di.xml:
```diff
@@ -43,6 +43,6 @@
     </type>
     <type name="Magento\Checkout\Controller\Cart">
-        <plugin name="multishipping_clear_addresses" type="Magento\Multishipping\Model\Cart\Controller\CartPlugin" />
+        <plugin name="multishipping_clear_addresses" type="Magento\Multishipping\Model\Cart\Controller\CartPlugin" sortOrder="50" />
```
@swnsma, looks pretty unreliable to me. Can you describe what is going on and maybe we will try to elaborate a robust fix together?

swnsma replied:
Hi @orlangur !
In two words, Inline Translate Phrase renderer loads later than error message Phrase become rendered. 

The error message source.
https://github.com/magento/magento2/blob/07bd6e560603aaca7d4088b241b5359fa08f4192/app/code/Magento/Bundle/Model/Product/Type.php#L1106
![error message source trace](https://www.tryimg.com/u/2018/07/13/errorMessageSource.png)

Rendered become loaded deep in `Magento\Framework\App\Action\Plugin\Design` plugin call.
The plugin has sort order 30.
https://github.com/magento/magento2/blob/07bd6e560603aaca7d4088b241b5359fa08f4192/app/code/Magento/Store/etc/di.xml#L68
![renderer load trace](https://www.tryimg.com/u/2018/07/13/setPhraseRendererStackTrace.png)

As alternative, sort order for this plugin could be set to 0, to prevent possible issue in future.

Commit:
magento/magento2#?: Fix Translation of error message on cart for deleted bundle option.
- change plugin sort order in case to prevent loading of not translatable messages for Localized exception in beforeDispatch plugin.

magento-engcom-team commented:
Hi @swnsma. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.7. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.
